test/unit/math/rev/mat/fun/cholesky_decompose_test.cpp

```diff
@@ -432,3 +437,22 @@ TEST(AgradRevMatrix, check_varis_on_stack_large) {
   X = stan::math::multiply(X, stan::math::transpose(X));
   test::check_varis_on_stack(stan::math::cholesky_decompose(X));
 }
+
+#ifdef STAN_OPENCL
+TEST(AgradRevMatrix, mat_cholesky_1st_deriv_large_gradients_opencl) {
+  stan::math::opencl_context.tuning_opts().cholesky_size_worth_transfer = 25;
```

Comment:
Can we add a test below the threshold too?

Reply:
Added a `test_gradient` and `test_gp_grad` under the threshold.